spec/system/provider_interface/provider_invites_user_spec.rb

```diff
@@ -72,4 +76,21 @@ def and_the_user_should_be_sent_a_welcome_email
     open_email(@email_address)
     expect(current_email.subject).to have_content t('provider_account_created.email.subject')
   end
+
+  def when_i_can_manage_users_for_another_provider
+    @provider_user.provider_permissions.find_by(provider: @another_provider).update(manage_users: true)
+  end
+
+  def and_i_add_a_provider_for_an_existing_user
+    fill_in 'First name', with: 'Jane'
+    fill_in 'Last name', with: 'Smith'
+    fill_in 'Email address', with: @email_address
+    check @another_provider.name_and_code
+    click_on 'Invite user'
+  end
+
+  def then_the_provider_is_assigned_to_the_user
+    expect(page).to have_content('Provider user invited')
+    expect(@new_provider_user.reload.providers).to include(@another_provider)
```

Comment:
If you add the following line the test fails:

```diff
+ expect(@new_provider_user.reload.providers).to include(@provider)
```

I think what you see here is that the invite overrides the original `providers` of the user because they're not in the form, so don't get passed to the form object.

Looking at this now, I suspect this bit will become quite tricky! With this feature and the edit feature you'll have to keep in mind that someone can have access to a completely unrelated provider, so the checked boxes in the form could be a subset of the providers a user should have access to.

Reply:
Yes I see what you mean, we need to populate the form with the providers from the existing user to reliably preserve them. This can't be done until we've submitted the form and discovered we're dealing with an existing user. 
If we just add the submitted providers to the existing user then there's no way to remove a provider (though that could arguably be handled in the _real_ edit form).

One fairly clunky way to get round the problem would be to return to the form in 'edit' mode populated with the existing user details but I'm not sure this would be clear to the managing user. I was thinking of an explicit message like:

_A user with this email already exists, if you would like to add them to specific providers please select them and resubmit this form._